### PR TITLE
Fixes custom image header templates

### LINKED ISSUE #2235: Editing HTML in the 'Image Header' Infowindow breaks the image
In an Infowindow set to Image Header, modifying the HTML causes the image to break, even if no code changes affect the header divs.

Steps to reproduce:

<strong>1. Create an Infowindow using Image Header (including creating the imageurl column, moving it to the top of the Infowindow fields, and turning it on.)</strong>
screen shot 2015-02-15 at 7 38 01 pm

<strong>2. Select and edit Custom HTML</strong>
screen shot 2015-02-15 at 7 39 04 pm

<strong>3. Hit Apply. Without having edited anything in the header divs or using {{imageurl}}, the image will now break.</strong>
screen shot 2015-02-15 at 7 39 20 pm

### PR DESCRIPTION
**Pull request by alonsogarciapablo**
Fixes #2235 and hides the `` tag when an infowindow doesn't have a valid cover URL (different issue).

image_infowindows

cc/ @cshirky

### CODE REVIEW

File: lib/assets/javascripts/cartodb/table/infowindow.js [hunk 1]
```diff
@@ -218,19 +218,18 @@
...
+      if (!this._containsCover()) return;
 
+      var $cover = this.$el.find(".cover");
+      var $imageNotFound = this.$el.find(".image_not_found");
+      var $img = $cover.find("img");
       var url = this._getCoverURL();
 
       if (!this._isValidURL(url)) {
         $imageNotFound.fadeIn(250);
+        $img.hide();
```
Comment (alonsogarciapablo) [inline lib/assets/javascripts/cartodb/table/infowindow.js]: We were not hiding the  tag before, and it was displaying a broken image (bonus)

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3286/))

**Comment by alonsogarciapablo** [comment]
Hey @xavijam, when you have a second, I'd love to have your thoughts on this PR. Thanks!

File: lib/assets/javascripts/cartodb/table/infowindow.js [hunk 2]
```diff
@@ -218,19 +218,18 @@
...
-      if (!this._containsCover()) return;
-
       var self = this;
 
-      var
-      $cover         = this.$el.find(".cover"),
-      $imageNotFound = this.$el.find(".image_not_found");
+      if (!this._containsCover()) return;
 
+      var $cover = this.$el.find(".cover");
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/table/infowindow.js]: ...this.$('.cover')...

File: lib/assets/javascripts/cartodb/table/infowindow.js [hunk 3]
```diff
@@ -218,19 +218,18 @@
...
+      var $imageNotFound = this.$el.find(".image_not_found");
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/table/infowindow.js]: Same here

**Comment by xavijam** [comment]
Shouldn't we change the same function in cartodb.js infowindow file?

**Comment by xavijam** [comment]
https://github.com/CartoDB/cartodb.js/blob/develop/src/geo/ui/infowindow.js#L558

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3323/))

**Comment by alonsogarciapablo** [comment]
@xavijam I fixed the selectors.

As per your other comment, I'd say that should also be changed in cartodb.js, but since nobody has complaint about it, I think we can wait.

**Comment by xavijam** [comment]
Open a ticket please :)

**Comment by alonsogarciapablo** [comment]
@xavijam: [There you have it](https://github.com/CartoDB/cartodb.js/issues/461). Thanks!

**Pull request merged**